### PR TITLE
PrintBGZFBlockInformation: remove file extension check so that we can accept, eg., bams

### LINKED ISSUE #5800: PrintBGZFBlockInformation should accept bam files
We just need to disable the unnecessary file extension check in this tool.

### PR DESCRIPTION
**Pull request by droazen**
The file extension check in this tool was redundant, since there is a subsequent check
that the input file is in BGZF format.

Resolves #5800

### CODE REVIEW

**Comment by droazen** [comment]
@jamesemery please review

File: src/test/java/org/broadinstitute/hellbender/tools/PrintBGZFBlockInformationIntegrationTest.java [hunk 1]
```diff
@@ -100,4 +100,20 @@ public void testRegularGzipFile() throws IOException {
         };
         runCommandLine(args);
     }
+
+    /* Make sure that we can handle a standard BAM file */
```
Comment (jamesemery) [inline src/test/java/org/broadinstitute/hellbender/tools/PrintBGZFBlockInformationIntegrationTest.java]: Maybe add a test to ensure there is a nice exception when the user provides a non-BGZF file (maybe a VCF).

File: src/main/java/org/broadinstitute/hellbender/tools/PrintBGZFBlockInformation.java [hunk 2]
```diff
@@ -71,11 +71,6 @@ protected void onStartup() {
             throw new UserException.CouldNotReadInputFile("File " + bgzfPathString + " does not exist");
         }
 
-        if ( ! IOUtil.hasBlockCompressedExtension(bgzfPathString) ) {
```
[the remark this answers is not in the document]
Comment (droazen) [reply src/main/java/org/broadinstitute/hellbender/tools/PrintBGZFBlockInformation.java]: Done and done

File: src/test/java/org/broadinstitute/hellbender/tools/PrintBGZFBlockInformationIntegrationTest.java (hunk 1 above)
Comment (droazen) [reply src/test/java/org/broadinstitute/hellbender/tools/PrintBGZFBlockInformationIntegrationTest.java]: Done

**Comment by droazen** [comment]
@jamesemery Comments addressed, back to you.

**Pull request merged**